docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,931 @@
+and finally using these certificates to authenticate to the domain as the domain
+administrator via Kerberos.
+
+Each certificate template vulnerability that will be discussed here has a ESC code, such
+as ESC1, ESC2, or ESC3. These ESC codes are taken from the original whitepaper that
+SpecterOps published which popularized these certificate template attacks, known as
+[Certified
+Pre-Owned](https://specterops.io/wp-content/uploads/sites/3/2022/06/Certified_Pre-Owned.pdf).
+In this paper Will Schroeder and Lee Christensen described 8 different domain escalation
+attacks that they found they could conduct via misconfigured certificate templates:
+
+- ESC1 - Domain escalation via No Issuance Requirements + Enrollable Client
+  Authentication/Smart Card Logon OID templates + CT_FLAG_ENROLLEE_SUPPLIES_SUBJECT
+- ESC2 - Domain escalation via No Issuance Requirements + Enrollable Any Purpose
+  EKU or no EKU
+- ESC3 - Domain escalation via No Issuance Requirements + Certificate Request
+  Agent EKU + no enrollment agent restrictions
+- ESC4 - Domain escalation via misconfigured certificate template access control
+- ESC5 - Domain escalation via vulnerable PKI AD Object Access Control
+- ESC6 - Domain escalation via the EDITF_ATTRIBUTESUBJECTALTNAME2 setting on CAs + No
+  Manager Approval + Enrollable Client Authentication/Smart Card Logon OID templates
+- ESC7 - Vulnerable Certificate Authority Access Control
+- ESC8 - NTLM Relay to AD CS HTTP Endpoints
+
+Later another
+[blog](https://research.ifcr.dk/certipy-4-0-esc9-esc10-bloodhound-gui-new-authentication-and-request-methods-and-more-7237d88061f7)
+came out from Oliver Lyak which discovered ESC9 and ESC10, two more vulnerabilities that
+could allow normal domain joined users to abuse certificate template misconfigurations to
+gain domain administrator privileges.
+
+- ESC9 - No Security Extension - CT_FLAG_NO_SECURITY_EXTENSION flag set in
+  `msPKI-EnrollmentFlag`. Also `StrongCertificateBindingEnforcement` not set to 2 or
+  `CertificateMappingMethods` contains `UPN` flag.
+- ESC10 - Weak Certificate Mappings -
+  `HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\SecurityProviders\Schannel
+  CertificateMappingMethods` contains `UPN` bit aka `0x4` or
+  `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\Kdc StrongCertificateBindingEnforcement` is set to `0`.
+
+Finally, we have ESC11, which was discovered by Compass Security and described in their
+[blog
+post](https://blog.compass-security.com/2022/11/relaying-to-ad-certificate-services-over-rpc/).
+
+- ESC11 - Relaying NTLM to ICPR - Relaying NTLM authentication to unprotected RPC
+  interface is allowed due to lack of the `IF_ENFORCEENCRYPTICERTREQUEST` flag on `Config.CA.Interface.Flags`.
+
+Currently Metasploit only supports attacking ESC1 to ESC3. Plans are in the works to add
+support for more ESC attacks in the future. As such this paper only covers exploiting ESC1
+to ESC3 at this time.
```

Comment:
An open source project saying they're going to do something rarely ages well. Let's just omit that detail. Maybe we'll get to it maybe we won't.

```suggestion
Currently Metasploit only supports attacking ESC1 to ESC3. As such this paper
only covers exploiting ESC1 to ESC3 at this time.
```

Reply:
Thanks, appreciate you calling this out!

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17399/commits/239bc42f4c4228a21dd726e08fa3273545112f7f